src/pkg/exporter/collector.go

```diff
@@ -0,0 +1,92 @@
+package exporter
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	namespace = "harbor"
+	subsystem = "exporter"
+)
+
+var (
+	scrapeDuration = typedDesc{
+		desc:      newDescWithLables(subsystem, "collector_duration_seconds", "Duration of a collector scrape", "collector"),
+		valueType: prometheus.GaugeValue,
+	}
+	scrapeSuccess = typedDesc{
+		desc:      newDescWithLables(subsystem, "collector_success", " Whether a collector succeeded.", "collector"),
+		valueType: prometheus.GaugeValue,
+	}
+)
+
+var hbrCli *HarborClient
+
+// HarborClient is client for request harbor
+type HarborClient struct {
```

Comment:
It seems so far all you need is http.Get and a URL constructor.

Reply:
Yep, But it's behavior is a harbor client